### PR TITLE
[SD-4206] - Targeted publishing

### PR DESCRIPTION
**Pull request by akintolga**
- [x] New settings page: Products
- [x] Ability to assign multiple products to subscribers
- [x] Add a *resend* functionality to published and corrected stories

### CODE REVIEW

**Comment by akintolga** [comment]
Depends on https://github.com/superdesk/superdesk-core/pull/259

File: scripts/superdesk-products/views/products-config-modal.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<div sd-modal data-model="modalActive" class="add-product-popup modal-large modal-static" >
```
Comment (ahilles107) [inline scripts/superdesk-products/views/products-config-modal.html]: indentation is completely messed up.

File: scripts/superdesk-products/views/settings.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<div sd-settings-view>
```
Comment (ahilles107) [inline scripts/superdesk-products/views/settings.html]: indentation is completely messed up.

**Comment by akintolga** [comment]
@ioanpocol @petrjasek FYI, once this is merged, we will need to create a product and assign it to subscribers to have the publishing to work. Products don't have to have a filter but subscribers have to have at least one product.
If you have a prepopulate script for NTB or PA for subscribers, it needs to change too. If you point those to me I'm happy to do that.

File: scripts/superdesk-authoring/metadata/metadata.js [hunk 3]
```diff
@@ -529,7 +529,7 @@ function MetaTermsDirective(metadata, $filter, $timeout) {
                 scope.tree = tree;
                 scope.activeTree = tree[null];
                 scope.combinedList = _.union(scope.list, scope.item[scope.field] ? scope.item[scope.field] : []);
-            });
+            }, true);
```
Comment (petrjasek) [inline scripts/superdesk-authoring/metadata/metadata.js]: imo this recursive watch might be an issue, list can be quite big right?

**Comment by petrjasek** [comment]
looks good :+1:

File: scripts/superdesk-authoring/metadata/metadata.js (hunk 3 above)
Comment (akintolga) [inline scripts/superdesk-authoring/metadata/metadata.js]: That's true. Removing...

File: scripts/superdesk-archive/directives.js [hunk 4]
```diff
@@ -759,6 +759,57 @@
...
+                            var data = {subscribers: getSubscriberIds(), version: scope.item._current_version};
+                            api.save('archive_resend', {}, data, scope.item)
+                                .then(function () {
+                                    notify.success(gettext('Item has been resent.'));
+                                    scope.cancel();
+                                }, function (response) {
+                                    if (response.data._message) {
+                                        notify.error(response.data._message);
+                                    } else {
+                                        notify.error(gettext('Unknown Error: Cannot kill the item'));
```
Comment (mdhaman) [inline scripts/superdesk-archive/directives.js]: Change the message to `Resend`.

File: scripts/superdesk-products/products.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        }])
+        .config(['apiProvider', function(apiProvider) {
+            apiProvider.api('products', {
+                type: 'http',
+                backend: {
+                    rel: 'products'
+                }
+            });
+        }])
+        .factory('products', ['$q', 'api', 'storage', 'contentFilters', function($q, api, storage, contentFilters) {
```
Comment (mdhaman) [inline scripts/superdesk-products/products.js]: `storage` not used.

**Comment by mdhaman** [comment]
:+1:  great job.

**Pull request merged**